pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -14,7 +21,115 @@ def setUpModule():  # pylint:disable=invalid-name
     utils.pulp_admin_login(config.get_config())
 
 
-class UploadDrpmTestCase(unittest.TestCase):
+class _BaseCLIDownloadTestCase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cfg = config.get_config()
+        cls.client = cli.Client(cls.cfg)
+
+    def temp_download(self, file_url, filename='tmp.rpm'):
+        """Download a file from the given url to a temp directory.
+
+        Args:
+            file_url: String. The URL of the file to download. Required
+            filename: String. Assigns a name to the file downloaded. Optional.
+
+        Returns:
+            String. Full path to file downloaded.
+
+        After the test case runs, the temp directory and file is destroyed.
+
+        """
+        temp_dir = self.client.run('mktemp --directory'.split()).stdout.strip()
+        self.addCleanup(self.client.run, 'rm -rf {}'.format(temp_dir).split())
+        file_path = os.path.join(temp_dir, os.path.split(filename)[-1])
+        try:
+            self.client.run(
+                'curl -o {} {}'.format(file_path, file_url).split()
+            )
+        except urllib.error.HTTPError:
+            raise self.skipTest('Skipping test, could not download resource')
```

Comment:
Maybe the exception should just bubble up and cause the test to fail? I don't see the advantage to obscuring an HTTPError.